src/components/transactions/GasStation/GasStation.tsx

```diff
@@ -42,28 +46,36 @@ export const GasStation: React.FC<GasStationProps> = ({
   skipLoad,
   disabled,
   rightComponent,
+  chainId,
 }) => {
-  const {
-    state,
-    gasPriceData: { data },
-  } = useGasStation();
-
-  const { walletBalances } = useWalletBalances();
+  const { state } = useGasStation();
+  const currentChainId = useRootStore((store) => store.currentChainId);
+  const selectedChainId = chainId ?? currentChainId;
```

Comment:
just a check that ?? only check against `null` and `undefined`.

Reply:
It's correct, if the user doesn't pass a chainId(undefined), it'll use the current chain id